sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/TransportClient.java

```diff
@@ -15,8 +17,13 @@
 // signature for backwards compatibility purposes.
 @SuppressWarnings("try")
 public abstract class TransportClient implements AutoCloseable {
+    private final boolean switchOffIOThreadForResponse;
     private ThroughputControlStore throughputControlStore;
 
+    public TransportClient() {
+        this.switchOffIOThreadForResponse = Configs.shouldSwitchOffIOThreadForResponse();
```

Comment:
Just want to state that with config in ctor , we will not able to change the logic on the fly, client has to be recycle . Which is ok given we are avoiding the cost of reading config on all request

Reply:
I don't think we need to read the config as part of the constructor since this is a final boolean. Why don't we initialize the value when declaring the variable ?

Reply:
That also works, changed to initialize the value when declaring the variable.